pages/index.vue

```diff
@@ -1,81 +1,46 @@
 <template>
-    <section class="container">
-      <div class="section-head">
-        <h4>{{ $t("Add Task") }}</h4>
+  <section class="container">
+    <div class="section-head">
+      <h4>{{ $t("Add Task") }}</h4>
+    </div>
+    <div class="btn-area">
+      <button @click="renderForm">{{ $t("Create") }}</button>
+      <div class="filter-area">
+        <FilterButtons />
       </div>
-      <div class="btn-area">
-        <button @click="showForm = true">{{ $t("Create") }}</button>
-        <div class="filter-area">
-            <button>{{ $t("All") }}</button>
-            <button>{{ $t("Incomplete") }}</button>
-            <button>{{ $t("Complete") }}</button>
-        </div>
-      </div>
-      <div class="task-area">
-        <div class="card" v-if="showForm">
-            <textarea v-model="todoText"></textarea>
-            <div class="card-footer">
-                <button @click="addTask"> {{ $t("Add Task") }} </button>
-                <span class="delete-icon">
-                    <img :src="deleteIcon" />
-                </span> 
-            </div>
-        </div>
-        <ToDo
-            v-for="todo in todoList"
-            :key="todo.id"
-            :todoItem="todo"
-        />
-      </div>
-      
-    </section>
+    </div>
+    <div class="task-area">
+      <AddTaskCard />
+      <ToDo v-for="todo in todoList" :key="todo.id" :todoItem="todo" />
+    </div>
+  </section>
 </template>
 
 <script>
-import { mapState } from 'vuex'
+import { mapState, mapMutations } from "vuex";
 export default {
-    layout: 'default',
-    computed: {
-        ...mapState('todo', [
-            'todoList'
-        ])
-    },
-    methods: {
-        addTask() {
-            if(!this.todoText) {
-                return
-            }
-           const createdAt = new Date().toISOString().substring(0,10);
-           const todo = {
-            text: this.todoText,
-            done: false,
-            createdAt: createdAt,
-            completedIn: null
-           }
-           this.$store.commit('todo/addTask', todo)
-           this.todoText = ''
-           this.showForm = false
-        }
-    },
-    data() {
-        return{
-            showForm: false,
-            todoText: '',
-            deleteIcon: require('@/assets/img/delete.svg')
-        }
+  layout: "default",
+  computed: {
+    ...mapState("todo", ["todoList"]),
+    ...mapMutations(["toggleForm"])
```

Comment:
mapMutations must be spread in `methods` property as mutation are functions. More [here](https://vuex.vuejs.org/guide/mutations.html#mutations-must-be-synchronous:~:text=essentially%20un%2Dtrackable!-,Committing%20Mutations%20in%20Components,-%23)

Reply:
resolved